typing/typecore.ml

```diff
@@ -6433,7 +6447,9 @@ and type_expect_
         type_label_access Legacy env srecord Env.Mutation lid in
       let ty_record =
         if expected_type = None
-        then newvar (Jkind.of_new_sort ~why:Record_assignment)
+        then newvar
+               (Jkind.of_new_sort ~why:Record_assignment
+                  ~level:(Ctype.get_current_level ()))
```

Comment:
I think this `then` clause typically goes on its own line, since it doesn't fit on the same line as the `then`.